### PR TITLE
Retry logic lose exeception bugfix

### PR DESCRIPTION
**Pull request by EvenLjj**
### Motivation:

Fix retry leads to lose exeception bug and sometime first time to call exeception bug

### Result:
Fixes
[issue 980](https://github.com/sofastack/sofa-rpc/issues/980)
[issue 984](https://github.com/sofastack/sofa-rpc/issues/984)

### CODE REVIEW

File: codec/codec-sofa-hessian/src/test/java/com/alipay/sofa/rpc/codec/sofahessian/SofaHessianSerializerTest.java [hunk 1]
```diff
@@ -81,6 +81,14 @@ public void encode() {
         Assert.assertTrue(error);
     }
 
+    @Test
```
Comment (OrezzerO) [inline codec/codec-sofa-hessian/src/test/java/com/alipay/sofa/rpc/codec/sofahessian/SofaHessianSerializerTest.java:84]: This test is not associate with other codes.

Comment (EvenLjj) [reply codec/codec-sofa-hessian/src/test/java/com/alipay/sofa/rpc/codec/sofahessian/SofaHessianSerializerTest.java]: fixed

**Review by OrezzerO — APPROVED** [review]
LGTM

**Comment by JervyShi** [comment]
Could you add some test cases for this issue?

**Comment by EvenLjj** [comment]
> Could you add some test cases for this issue?

OK

**Comment by JervyShi** [comment]
LGTM

**Review by JervyShi — APPROVED** [review]
LGTM

**Pull request merged**